Upgrade React

kckst8 opened the pull request:
v17, to inline with iTwin.js.....this will publish new major versions of these packages as this is a potentially breaking change

aruniverse commented:
Should this really be a major bump? 
We do need to take these to 1.0 soon but I was thinking we'd refactor it to use itwin instead of project/context. 

I was thinking a minor hump would suffice since this is already at 0.x

Also were there no types/react react-dom pkgs to update? Maybe those deps aren't used (didn't see them on the diff on the mobile version)

kckst8 commented:
> 

this is definitely a major bump (it will break consumers of React 16 for example). Good point about renaming contextId, but are we certain that all of those apis have been renamed? These ultimately should mirror the apis in terms

bentleyvk commented:
Do we really need to make minimum React version 17? A lot of libraries (including iTwinUI) allows to use v16 and v17:
![image](https://user-images.githubusercontent.com/36186912/151171867-e022d828-74b2-43f6-9001-931c1698a866.png)
Because v17 shouldn't really break anything. Also, I agree with Arun that since we are in 0.x, it should be fine.

kckst8 commented:
> Do we really need to make minimum React version 17? A lot of libraries (including iTwinUI) allow to use v16 and v17: ![image](https://user-images.githubusercontent.com/36186912/151171867-e022d828-74b2-43f6-9001-931c1698a866.png) Because v17 shouldn't really break anything. Also, I agree with Arun that since we are in 0.x, it should be fine.

Maybe using or would be the best bet here. That would not be a breaking change. However, as it stands right now it is definitely a breaking change and thus a major bump. I'll change to or

kckst8 commented:
> > Do we really need to make minimum React version 17? A lot of libraries (including iTwinUI) allow to use v16 and v17: ![image](https://user-images.githubusercontent.com/36186912/151171867-e022d828-74b2-43f6-9001-931c1698a866.png) Because v17 shouldn't really break anything. Also, I agree with Arun that since we are in 0.x, it should be fine.
> 
> Maybe using or would be the best bet here. That would not be a breaking change. However, as it stands right now it is definitely a breaking change and thus a major bump. I'll change to or

Updated to 16 || 17 and changed to a patch update

aruniverse commented:
> Updated to 16 || 17 and changed to a patch update

peer dep looks fine; im surprised you can use || for dev dep; is that part right?

kckst8 commented:
> > Updated to 16 || 17 and changed to a patch update
> 
> peer dep looks fine; im surprised you can use || for dev dep; is that part right?

nah find/replace mishap....to answer it will work (takes the greater I believe), but doesn't make much sense. I updated them

aruniverse reviewed (approved):
Approving, i think this could be a minor bump instead, but wont hold the pr

raplemie commented on packages/modules/imodel-browser/src/utils/_buildMenuOptions.tsx:
```diff
@@ -9,7 +9,7 @@ export interface ContextMenuBuilderItem<T = any>
   key: string;
   visible?: boolean | ((value: T) => boolean);
-  onClick?: (value: T) => void;
+  onClick?: ((value?: unknown) => void) | undefined;
```
MenuItem changed. Ok.

raplemie reviewed (approved):
LGTM
However, I didnt dig that much into the changes.
